enabler/src/de/schildbach/pte/AbstractOpenTripPlannerProvider.java

```diff
@@ -0,0 +1,803 @@
+ *     </li>
+ * </ul>
+ * @author Holger Bruch
+ * @author Kavitha Ravi
+ */
+class AbstractOpenTripPlannerProvider extends AbstractNetworkProvider {
+
+    protected final static String SERVER_PRODUCT = "otp";
+    protected final static String SERVER_VERSION = "v1.2";
+    private final String router;
+    private final Multimap<Product, String> productsToModeMap ;
+    private Locale locale = Locale.ENGLISH;
+
+    enum OTPMode {
+        WALK, BIKE, CAR, RAIL, SUBWAY, TRAM, BUS, CABLE_CAR, FERRY, GONDOLA, FUNICULAR, AIRPLANE,
+        TRANSIT
+    }
+
+    protected final HttpUrl apiBase ;
+    
+    @SuppressWarnings("serial")
+    private static class Context implements QueryTripsContext {
+        @Override
+        public boolean canQueryLater() {
+            return false;
+        }
+
+        @Override
+        public boolean canQueryEarlier() {
+            return false;
+        }
+    }
+
+    protected AbstractOpenTripPlannerProvider(NetworkId network, HttpUrl apiBase) {
+        this(network, apiBase, "default", null);
+    }
+
+    protected AbstractOpenTripPlannerProvider(NetworkId network, HttpUrl apiBase, String router, Multimap<Product, String> productsToMode) {
+        super(network);
+        this.apiBase = apiBase;
+        if (productsToMode == null) {
+            this.productsToModeMap = defaultProductToOTPModeMapping();
+        } else {
+            this.productsToModeMap = productsToMode;
+        }
+        this.router = router;
+        this.httpClient.setHeader("Accept", "application/json");
+    }
+
+    private HttpUrl.Builder url() {
+        return apiBase.newBuilder().addPathSegment("routers").addPathSegment(router());
+    }
+
+    protected String router() {
+        return router;
+    }
+
+    @Override
+    protected boolean hasCapability(final Capability capability) {
+        if (capability == Capability.SUGGEST_LOCATIONS || capability == Capability.NEARBY_LOCATIONS
+                || capability == Capability.DEPARTURES || capability == Capability.TRIPS)
+            return true;
+        else
+            return false;
+    }
+
+    /**
+     * Set {@code Locale} used to query te OTP server.
+     * @param locale
+     */
+    public void setLocale(Locale locale){
+        this.locale = locale;
+    }
+
+    protected Multimap<Product, String> defaultProductToOTPModeMapping() {
+        Multimap<Product, String> productsToModeMap = ArrayListMultimap.create();
+        productsToModeMap.put(Product.HIGH_SPEED_TRAIN, OTPMode.RAIL.toString());
+        productsToModeMap.put(Product.REGIONAL_TRAIN, OTPMode.RAIL.toString());
+        productsToModeMap.put(Product.SUBURBAN_TRAIN, OTPMode.RAIL.toString());
+        productsToModeMap.put(Product.SUBWAY, OTPMode.SUBWAY.toString());
+        productsToModeMap.put(Product.TRAM, OTPMode.TRAM.toString());
+        // GONDOLA and FUNICULAR are no PTE Products, but OTP modes. Associate them with TRAM (?)
```

Comment:
Ok, I agree, FUNICULAR might be seen as a specialization of cable car. But for gondolas PTE has no corresponding product. For OTP, [gondolas correspond to telecabin services](https://github.com/opentripplanner/OpenTripPlanner/blob/f60440f12f9465390bc39dea7ee8e120d382eb3f/src/main/java/org/opentripplanner/gtfs/GtfsLibrary.java#L122), for which PTE has currently no product. 
For lines, the product is not mandatory and may be null when parsing an OTP response in such a case. I guess I should really set it to null for gondolas.
For requests, I think it should be subsumed under a more or less resembling code, so it can be (de)selected when filtering products.